Mustachio/Tokenizer.cs

```diff
@@ -218,18 +216,25 @@ public static TokenizeResult Tokenize(string templateString, ParsingOptions pars
             return new TokenizeResult { Tokens = tokens, Errors = parseErrors };
         }
 
-        private static void TokenizeCustomExpander(IEnumerable<TokenExpander> expanders, Match m, ParsingOptions options,
+        private static bool DidTokenizeCustomExpander(ILookup<Precedence, TokenExpander> expanders, Precedence precedence, Match m, ParsingOptions options,
             ref List<TokenTuple> tokens, ref List<IndexedParseException> parseErrors)
         {
-            var expander = expanders.First(e => Regex.IsMatch(m.Value, e.RegEx));
+            var expander = expanders[precedence].FirstOrDefault(e => Regex.IsMatch(m.Value, e.RegEx));
+            if (expander == null)
+            {
+                return false;
+            }
             if (expander.ExpandTokens == null)
             {
                 throw new ArgumentException($"ExpandTokens function was not provided for expander with RegEx: {expander.RegEx}");
```

Comment:
I'm not sure if this is correct.

Token expanders can either be expansions here, or rendering later, or both. It looks like you're adding the custom token to the stream and also expanding it, but the two operations are not dependent upon each other. It is only important that at least expansion, or a renderer is specified, but the existence of each determines whether the custom token should be emitted, or expansion results be added.

Reply:
I agree it is more correct to not emit the custom token if the renderer is not set and that they are in a way separate operations. 
I chose not to throw an exception if both are unset, but if this validation is useful then I can add it.